components/announcement_bar/announcement_bar_controller.tsx

```diff
@@ -2,37 +2,36 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
+
+import {ClientLicense, ClientConfig, WarnMetricStatus} from 'mattermost-redux/types/config';
+import {Dictionary} from 'mattermost-redux/types/utilities';
 
 import ConfigurationAnnouncementBar from './configuration_bar';
 import VersionBar from './version_bar';
-import TextDismissableBar from './text_dismissable_bar.jsx';
+import TextDismissableBar from './text_dismissable_bar';
 import AnnouncementBar from './default_announcement_bar';
 
 import CloudAnnouncementBar from './cloud_announcement_bar';
 import PaymentAnnouncementBar from './payment_announcement_bar';
 import CloudTrialAnnouncementBar from './cloud_trial_announcement_bar';
 
-export default class AnnouncementBarController extends React.PureComponent {
-    static propTypes = {
-        license: PropTypes.object,
-        config: PropTypes.object,
-        user: PropTypes.shape({
-            email: PropTypes.string.isRequired,
-            email_verified: PropTypes.bool,
-        }),
-        canViewSystemErrors: PropTypes.bool.isRequired,
-        latestError: PropTypes.object,
-        totalUsers: PropTypes.number,
-        warnMetricsStatus: PropTypes.object,
-        actions: PropTypes.shape({
-            dismissError: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    license?: ClientLicense;
+    config?: Partial<ClientConfig>;
+    canViewSystemErrors: boolean;
+    latestError?: {
+        error: any;
+    };
+    warnMetricsStatus?: Dictionary<WarnMetricStatus>;
+    actions: {
+        dismissError: (index: number) => void;
+    };
+}
 
+export default class AnnouncementBarController extends React.PureComponent<Props> {
     render() {
         let adminConfiguredAnnouncementBar = null;
-        if (this.props.config.EnableBanner === 'true' && this.props.config.BannerText.trim()) {
+        if (this.props.config && this.props.config.EnableBanner === 'true' && this.props.config.BannerText?.trim()) {
```

Comment:
nit: Optional chaining may also work here to reduce the checks

```suggestion
        if (this.props.config?.EnableBanner === 'true' && this.props.config.BannerText?.trim()) {
```